src/odemis/acq/stream/_projection.py

```diff
@@ -1300,19 +1300,21 @@ def _on_selected_line(self, _):
     def _on_selected_time(self, _):
         self._shouldUpdateImage()
 
+    def _find_metadata(self, md):
+        return md  # The data from _computeSpec() should already be correct
+
     def _computeSpec(self):
         """
         Compute the 1D spectrum from the stream.calibrated VA using the
         selected_time, selected_line, and width.
 
-        return: spec1d, md
-            where spec1d is the line spectrum (1D array)
-            and md is the metadata structure
+        return DataArray of of shape XC: the line spectrum
+           the distance increases in X and the wavelength increases in C
         """
 
         if ((None, None) in self.stream.selected_line.value or
                 self.stream.calibrated.value.shape[0] == 1):
-            return None, None
+            return None
```

Comment:
If the expected return value is a tuple, will this work?

Reply:
I've updated all the (2) callers, and changed all the places with a return. So it should work.